code/Items/ItemFilter.cs

```diff
@@ -11,36 +11,36 @@ public class ItemFilter
         /// <summary>
         /// Not authorized items
         /// </summary>
-        public HashSet<Item> Blacklist = new();
+        public HashSet<string> Blacklist = new();
 
         /// <summary>
         /// Authorized items
         /// </summary>
-        public HashSet<Item> Whitelist = new();
+        public HashSet<string> Whitelist = new();
 
-        public bool IsAuthorized(Item item)
+        public bool IsAuthorized(string itemId)
```

Comment:
When you upgrade a class like that , you can let the old way.
You can add by id or by item. The item use the IsAuthorized(item.Id)

Reply:
(I resolved the other comments to talk only here)
I don't think using the Item for the filter is a good practice since Item = Entity but I agree that IsAuthorized could accept an Item and only get its id

Reply:
I think we can add by string , by itemData and by item. if you don't like, don't do it i'm fine without for now

Reply:
Adding or checking ?

Reply:
All